File: dotnet/GitHub/Repos/Item/Item/WithRepoItemRequestBuilder.cs
using GitHubTodoDemo.GitHub.Repos.Item.Item.Commits;
using GitHubTodoDemo.GitHub.Repos.Item.Item.Pulls;
using Microsoft.Kiota.Abstractions;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using System;
namespace GitHubTodoDemo.GitHub.Repos.Item.Item {
    /// <summary>
    /// Builds and executes requests for operations under \repos\{owner}\{repo}
    /// </summary>
    public class WithRepoItemRequestBuilder : BaseRequestBuilder {
        /// <summary>The commits property</summary>
        public CommitsRequestBuilder Commits { get =>
            new CommitsRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The pulls property</summary>
        public PullsRequestBuilder Pulls { get =>
            new PullsRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>
        /// Instantiates a new WithRepoItemRequestBuilder and sets the default values.
        /// </summary>
        /// <param name="pathParameters">Path parameters for the request</param>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        public WithRepoItemRequestBuilder(Dictionary<string, object> pathParameters, IRequestAdapter requestAdapter) : base(requestAdapter, "{+baseurl}/repos/{owner}/{repo}", pathParameters) {
        }
        /// <summary>
        /// Instantiates a new WithRepoItemRequestBuilder and sets the default values.
        /// </summary>
        /// <param name="rawUrl">The raw URL to use for the request builder.</param>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        public WithRepoItemRequestBuilder(string rawUrl, IRequestAdapter requestAdapter) : base(requestAdapter, "{+baseurl}/repos/{owner}/{repo}", rawUrl) {
        }
    }
}

File: php/GitHub/Repos/Item/Item/WithRepoItemRequestBuilder.php
<?php

namespace Baywet\Githubtododemo\Github\Repos\Item\Item;

use Baywet\Githubtododemo\Github\Repos\Item\Item\Commits\CommitsRequestBuilder;
use Baywet\Githubtododemo\Github\Repos\Item\Item\Pulls\PullsRequestBuilder;
use Microsoft\Kiota\Abstractions\BaseRequestBuilder;
use Microsoft\Kiota\Abstractions\RequestAdapter;

/**
 * Builds and executes requests for operations under /repos/{owner}/{repo}
*/
class WithRepoItemRequestBuilder extends BaseRequestBuilder 
{
    /**
     * The commits property
    */
    public function commits(): CommitsRequestBuilder {
        return new CommitsRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * The pulls property
    */
    public function pulls(): PullsRequestBuilder {
        return new PullsRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * Instantiates a new WithRepoItemRequestBuilder and sets the default values.
     * @param array<string, mixed>|string $pathParametersOrRawUrl Path parameters for the request or a String representing the raw URL.
     * @param RequestAdapter $requestAdapter The request adapter to use to execute the requests.
    */
    public function __construct($pathParametersOrRawUrl, RequestAdapter $requestAdapter) {
        parent::__construct($requestAdapter, [], '{+baseurl}/repos/{owner}/{repo}');
        if (is_array($pathParametersOrRawUrl)) {
            $this->pathParameters = $pathParametersOrRawUrl;
        } else {
            $this->pathParameters = ['request-raw-url' => $pathParametersOrRawUrl];
        }
    }

}
